### PR TITLE
Feature - Multiple organization support

### PR DESCRIPTION
**Pull request by gsmithun4**
Resolves https://github.com/ToolJet/ToolJet/issues/2777

Multi organization - https://multi-org-tj.herokuapp.com/
Single organization - https://single-org-tj.herokuapp.com/

### CODE REVIEW

File: server/src/entities/sso_config.entity.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  id: string;
+
+  @Column({ name: 'organization_id' })
+  organizationId: string;
+
+  @Column({ name: 'sso' })
+  sso: 'google' | 'git' | 'okta' | 'form';
+
+  @Column({ type: 'json' })
+  configs: Google | Git | Okta | Form | any;
```
Comment (akshaysasidrn) [inline server/src/entities/sso_config.entity.ts:43]: Nice! We should also be encrypting the client creds as well.

**Review by akshaysasidrn — CHANGES_REQUESTED** [review]

File: server/migrations/1648589182504-PopulateSSOConfigs.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+export class PopulateSSOConfigs1648589182504 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const entityManager = queryRunner.manager;
+    const OrganizationRepository = entityManager.getRepository(Organization);
+
+    const organizations: Organization[] = await OrganizationRepository.find({ relations: ['ssoConfigs'] });
+
+    if (organizations && organizations.length > 0) {
+      for (const organization of organizations) {
+        // adding form configs for organizations which does not have any
```
[the remark this answers is not in the document]
Comment (akshaysasidrn) [reply server/migrations/1648589182504-PopulateSSOConfigs.ts:14]: What happens for new org registration, won't they be empty?

File: server/migrations/1645864719155-MultiOrganisation.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.addColumn(
+      'organization_users',
+      new TableColumn({
+        name: 'invitation_token',
+        type: 'varchar',
+        isNullable: true,
+      })
+    );
+    await queryRunner.dropColumn('users', 'sso');
```
[the remark this answers is not in the document]
Comment (gsmithun4) [reply server/migrations/1645864719155-MultiOrganisation.ts:13]: Yes, this column is not used anymore, we were inserting mode of login here.

File: server/migrations/1648589182504-PopulateSSOConfigs.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gsmithun4) [reply server/migrations/1648589182504-PopulateSSOConfigs.ts:14]: we have to enable password login for existing organizations. for new organizations we are doing the same, password login is enabled by default. 
But admin should configure the SSO details, we are not using environment variables anymore. Should we read the environment variables and insert into tables?

File: server/src/entities/user.entity.ts [hunk 4]
```diff
@@ -89,4 +76,7 @@ export class User extends BaseEntity {
 
   @OneToMany(() => UserGroupPermission, (userGroupPermission) => userGroupPermission.user, { onDelete: 'CASCADE' })
   userGroupPermissions: UserGroupPermission[];
+
+  organizationId: string;
+  isFormLogin: boolean;
```
[the remark this answers is not in the document]
Comment (gsmithun4) [reply server/src/entities/user.entity.ts:81]: will go with passwordLogin

File: server/src/entities/sso_config.entity.ts (hunk 1 above)
Comment (gsmithun4) [reply server/src/entities/sso_config.entity.ts:43]: we are encrypting any keys containing `secret`

File: server/migrations/1648589182504-PopulateSSOConfigs.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (akshaysasidrn) [reply server/migrations/1648589182504-PopulateSSOConfigs.ts]: Got it. So yeah, I'd think we'd have to do that as well for existing SSO users as a part of migration.

[the remark this answers is not in the document]
Comment (gsmithun4) [reply server/migrations/1648589182504-PopulateSSOConfigs.ts]: Done

**Review by Navaneeth-pk — APPROVED** [review]

**Pull request merged**